mantidproject/mantid
# Replace uses of deprecated `CheckWorkspacesMatch` with `CompareWorkspaces`

See the title.

**To test:**

1. Make sure I have not missed any other appropriate occurrence of `CheckWorkspacesMatch`.
2. Pick a system test that uses built-in validation with `validate` method. Introduce a change in the test, such that it should fail. Run and make sure it fails.



Fixes #17704.

Does not need to be in the release notes.



---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
